File: php-frontend/classes/ResponseHandler.php
<?php
/**
 * 统一响应处理类
 * 提供标准化的API响应格式
 */
class ResponseHandler {
    
    /**
     * 成功响应
     */
    public static function success($data = null, $message = '操作成功', $code = 200) {
        $response = [
            'success' => true,
            'message' => $message,
            'code' => $code,
            'timestamp' => date('Y-m-d H:i:s'),
            'data' => $data
        ];
        
        self::sendResponse($response, $code);
    }
    
    /**
     * 错误响应
     */
    public static function error($message = '操作失败', $code = 400, $errors = null) {
        $response = [
            'success' => false,
            'message' => $message,
            'code' => $code,
            'timestamp' => date('Y-m-d H:i:s'),
            'errors' => $errors
        ];
        
        self::sendResponse($response, $code);
    }
    
    /**
     * 验证错误响应
     */
    public static function validationError($errors, $message = '验证失败') {
        self::error($message, 422, $errors);
    }
    
    /**
     * 未授权响应
     */
    public static function unauthorized($message = '未授权访问') {
        self::error($message, 401);
    }
    
    /**
     * 禁止访问响应
     */
    public static function forbidden($message = '权限不足') {
        self::error($message, 403);
    }
    
    /**
     * 未找到响应
     */
    public static function notFound($message = '资源未找到') {
        self::error($message, 404);
    }
    
    /**
     * 服务器错误响应
     */
    public static function serverError($message = '服务器内部错误') {
        self::error($message, 500);
    }
    
    /**
     * 分页响应
     */
    public static function paginated($data, $pagination, $message = '获取成功') {
        $response = [
            'success' => true,
            'message' => $message,
            'code' => 200,
            'timestamp' => date('Y-m-d H:i:s'),
            'data' => $data,
            'pagination' => $pagination
        ];
        
        self::sendResponse($response, 200);
    }
    
    /**
     * 发送响应
     */
    private static function sendResponse($response, $httpCode) {
        // 设置HTTP状态码
        http_response_code($httpCode);
        
        // 设置响应头
        header('Content-Type: application/json; charset=utf-8');
        header('Cache-Control: no-cache, must-revalidate');
        header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
        
        // 输出JSON响应
        echo json_encode($response, JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT);
        exit;
    }
    
    /**
     * 重定向响应
     */
    public static function redirect($url, $message = null) {
        if ($message) {
            $_SESSION['flash_message'] = $message;
        }
        
        header('Location: ' . $url);
        exit;
    }
    
    /**
     * 显示消息页面
     */
    public static function showMessage($title, $message, $type = 'info', $redirectUrl = null) {
        $pageTitle = $title;
        $showSidebar = isset($_SESSION['user']);
        
        include 'views/layout/header.php';
        
        echo '<div class="container mt-5">';
        echo '<div class="row justify-content-center">';
        echo '<div class="col-md-8">';
        echo '<div class="card shadow-lg border-0">';
        echo '<div class="card-body text-center p-5">';
        
        // 图标
        $iconClass = match($type) {
            'success' => 'bi-check-circle text-success',
            'error' => 'bi-exclamation-triangle text-danger',
            'warning' => 'bi-exclamation-triangle text-warning',
            'info' => 'bi-info-circle text-info',
            default => 'bi-info-circle text-info'
        };
        
        echo '<div class="mb-4">';
        echo '<i class="bi ' . $iconClass . '" style="font-size: 5rem;"></i>';
        echo '</div>';
        
        echo '<h1 class="display-4 mb-3">' . htmlspecialchars($title) . '</h1>';
        echo '<p class="lead text-muted mb-4">' . htmlspecialchars($message) . '</p>';
        
        if ($redirectUrl) {
            echo '<div class="d-grid gap-2 d-md-flex justify-content-md-center">';
            echo '<a href="' . htmlspecialchars($redirectUrl) . '" class="btn btn-primary btn-lg">';
            echo '<i class="bi bi-arrow-left me-2"></i>返回';
            echo '</a>';
            echo '</div>';
        }
        
        echo '</div>';
        echo '</div>';
        echo '</div>';
        echo '</div>';
        echo '</div>';
        
        include 'views/layout/footer.php';
        exit;
    }
    
    /**
     * 显示错误页面
     */
    public static function showError($title, $message, $details = null) {
        $_SESSION['error_title'] = $title;
        $_SESSION['error_message'] = $message;
        $_SESSION['error_data'] = $details;
        
        header('Location: /error');
        exit;
    }
    
    /**
     * 显示JSON错误响应
     */
    public static function jsonError($message, $code = 400, $errors = null) {
        $response = [
            'success' => false,
            'message' => $message,
            'code' => $code,
            'timestamp' => date('Y-m-d H:i:s'),
            'errors' => $errors
        ];
        
        http_response_code($code);
        header('Content-Type: application/json; charset=utf-8');
        echo json_encode($response, JSON_UNESCAPED_UNICODE);
        exit;
    }
    
    /**
     * 显示JSON成功响应
     */
    public static function jsonSuccess($data = null, $message = '操作成功') {
        $response = [
            'success' => true,
            'message' => $message,
            'code' => 200,
            'timestamp' => date('Y-m-d H:i:s'),
            'data' => $data
        ];
        
        http_response_code(200);
        header('Content-Type: application/json; charset=utf-8');
        echo json_encode($response, JSON_UNESCAPED_UNICODE);
        exit;
    }
    
    /**
     * 处理异常
     */
    public static function handleException($exception) {
        $message = $exception->getMessage();
        $code = $exception->getCode() ?: 500;
        
        // 记录错误日志
        ErrorHandler::logCustomError($message, [
            'file' => $exception->getFile(),
            'line' => $exception->getLine(),
            'trace' => $exception->getTraceAsString()
        ]);
        
        // 根据请求类型返回响应
        if (self::isAjaxRequest()) {
            self::jsonError($message, $code);
        } else {
            self::showError('系统错误', $message);
        }
    }
    
    /**
     * 检查是否为AJAX请求
     */
    private static function isAjaxRequest() {
        return !empty($_SERVER['HTTP_X_REQUESTED_WITH']) && 
               strtolower($_SERVER['HTTP_X_REQUESTED_WITH']) === 'xmlhttprequest';
    }
    
    /**
     * 获取分页信息
     */
    public static function getPagination($page, $pageSize, $total) {
        $totalPages = ceil($total / $pageSize);
        
        return [
            'current_page' => $page,
            'page_size' => $pageSize,
            'total_items' => $total,
            'total_pages' => $totalPages,
            'has_next' => $page < $totalPages,
            'has_prev' => $page > 1,
            'next_page' => $page < $totalPages ? $page + 1 : null,
            'prev_page' => $page > 1 ? $page - 1 : null
        ];
    }
    
    /**
     * 格式化文件大小
     */
    public static function formatFileSize($bytes) {
        $units = ['B', 'KB', 'MB', 'GB', 'TB'];
        $bytes = max($bytes, 0);
        $pow = floor(($bytes ? log($bytes) : 0) / log(1024));
        $pow = min($pow, count($units) - 1);
        
        $bytes /= pow(1024, $pow);
        
        return round($bytes, 2) . ' ' . $units[$pow];
    }
    
    /**
     * 格式化时间
     */
    public static function formatTime($timestamp) {
        $time = time() - $timestamp;
        
        if ($time < 60) {
            return '刚刚';
        } elseif ($time < 3600) {
            return floor($time / 60) . '分钟前';
        } elseif ($time < 86400) {
            return floor($time / 3600) . '小时前';
        } elseif ($time < 2592000) {
            return floor($time / 86400) . '天前';
        } else {
            return date('Y-m-d H:i:s', $timestamp);
        }
    }
}
?>

File: backend/app/core/response_handler.py
"""
统一响应处理器
提供标准化的API响应格式，匹配模拟API的格式（success/data/message）
"""

from typing import Dict, Any, Optional, Union, List
from fastapi import Response
from fastapi.responses import JSONResponse
from datetime import datetime
import json


class ResponseHandler:
    """统一响应处理器"""
    
    @staticmethod
    def success(
        data: Optional[Any] = None,
        message: str = "操作成功",
        status_code: int = 200,
        additional_data: Optional[Dict[str, Any]] = None
    ) -> JSONResponse:
        """
        成功响应
        
        Args:
            data: 响应数据
            message: 响应消息
            status_code: HTTP状态码
            additional_data: 额外的响应数据
            
        Returns:
            JSONResponse: 格式化的成功响应
        """
        response = {
            "success": True,
            "message": message,
            "timestamp": datetime.now().isoformat()
        }
        
        # 添加数据
        if data is not None:
            response["data"] = data
        
        # 添加额外数据
        if additional_data:
            response.update(additional_data)
        
        return JSONResponse(
            status_code=status_code,
            content=response
        )
    
    @staticmethod
    def error(
        message: str = "操作失败",
        status_code: int = 400,
        error_code: Optional[str] = None,
        details: Optional[Dict[str, Any]] = None,
        errors: Optional[List[Dict[str, Any]]] = None
    ) -> JSONResponse:
        """
        错误响应
        
        Args:
            message: 错误消息
            status_code: HTTP状态码
            error_code: 错误代码
            details: 错误详情
            errors: 验证错误列表
            
        Returns:
            JSONResponse: 格式化的错误响应
        """
        response = {
            "success": False,
            "message": message,
            "timestamp": datetime.now().isoformat()
        }
        
        # 添加错误代码
        if error_code:
            response["error_code"] = error_code
        
        # 添加错误详情
        if details:
            response["details"] = details
        
        # 添加验证错误
        if errors:
            response["errors"] = errors
        
        return JSONResponse(
            status_code=status_code,
            content=response
        )
    
    @staticmethod
    def paginated(
        data: List[Any],
        total: int,
        page: int,
        size: int,
        message: str = "获取成功"
    ) -> JSONResponse:
        """
        分页响应
        
        Args:
            data: 数据列表
            total: 总记录数
            page: 当前页码
            size: 每页大小
            message: 响应消息
            
        Returns:
            JSONResponse: 格式化的分页响应
        """
        total_pages = (total + size - 1) // size if size > 0 else 0
        
        return ResponseHandler.success(
            data=data,
            message=message,
            additional_data={
                "pagination": {
                    "total": total,
                    "page": page,
                    "size": size,
                    "total_pages": total_pages,
                    "has_next": page < total_pages,
                    "has_prev": page > 1
                }
            }
        )
    
    @staticmethod
    def created(
        data: Optional[Any] = None,
        message: str = "创建成功"
    ) -> JSONResponse:
        """
        创建成功响应
        
        Args:
            data: 创建的数据
            message: 响应消息
            
        Returns:
            JSONResponse: 格式化的创建成功响应
        """
        return ResponseHandler.success(
            data=data,
            message=message,
            status_code=201
        )
    
    @staticmethod
    def updated(
        data: Optional[Any] = None,
        message: str = "更新成功"
    ) -> JSONResponse:
        """
        更新成功响应
        
        Args:
            data: 更新的数据
            message: 响应消息
            
        Returns:
            JSONResponse: 格式化的更新成功响应
        """
        return ResponseHandler.success(
            data=data,
            message=message
        )
    
    @staticmethod
    def deleted(
        message: str = "删除成功"
    ) -> JSONResponse:
        """
        删除成功响应
        
        Args:
            message: 响应消息
            
        Returns:
            JSONResponse: 格式化的删除成功响应
        """
        return ResponseHandler.success(
            message=message
        )
    
    @staticmethod
    def not_found(
        resource: str = "资源",
        identifier: str = ""
    ) -> JSONResponse:
        """
        资源未找到响应
        
        Args:
            resource: 资源名称
            identifier: 资源标识符
            
        Returns:
            JSONResponse: 格式化的未找到响应
        """
        message = f"{resource}未找到"
        if identifier:
            message += f": {identifier}"
            
        return ResponseHandler.error(
            message=message,
            status_code=404,
            error_code="NOT_FOUND"
        )
    
    @staticmethod
    def validation_error(
        errors: List[Dict[str, Any]],
        message: str = "验证失败"
    ) -> JSONResponse:
        """
        验证错误响应
        
        Args:
            errors: 验证错误列表
            message: 响应消息
            
        Returns:
            JSONResponse: 格式化的验证错误响应
        """
        return ResponseHandler.error(
            message=message,
            status_code=422,
            error_code="VALIDATION_ERROR",
            errors=errors
        )
    
    @staticmethod
    def unauthorized(
        message: str = "未授权访问"
    ) -> JSONResponse:
        """
        未授权响应
        
        Args:
            message: 响应消息
            
        Returns:
            JSONResponse: 格式化的未授权响应
        """
        return ResponseHandler.error(
            message=message,
            status_code=401,
            error_code="UNAUTHORIZED"
        )
    
    @staticmethod
    def forbidden(
        message: str = "权限不足"
    ) -> JSONResponse:
        """
        权限不足响应
        
        Args:
            message: 响应消息
            
        Returns:
            JSONResponse: 格式化的权限不足响应
        """
        return ResponseHandler.error(
            message=message,
            status_code=403,
            error_code="FORBIDDEN"
        )
    
    @staticmethod
    def server_error(
        message: str = "服务器内部错误",
        details: Optional[Dict[str, Any]] = None
    ) -> JSONResponse:
        """
        服务器错误响应
        
        Args:
            message: 响应消息
            details: 错误详情
            
        Returns:
            JSONResponse: 格式化的服务器错误响应
        """
        return ResponseHandler.error(
            message=message,
            status_code=500,
            error_code="INTERNAL_SERVER_ERROR",
            details=details
        )


# 创建默认实例
response_handler = ResponseHandler()

# 导出主要组件
__all__ = [
    "ResponseHandler",
    "response_handler"
]